### PR TITLE
Ensuring MatrixConstructor works for Firefox

### PR DESCRIPTION
**Pull request by Antonio-Laguna**
Currently the MatrixConstructor doesn't really work with Firefox, this fixes the issue.

### CODE REVIEW

**Comment by Belelros** [comment]
I signed it!

**Comment by Belelros** [comment]
I signed it!

**Comment by concavelenz** [comment]
Thanks for the PR, can you enable the appropriate tests in style/transform_test.js?

**Comment by Belelros** [comment]
Sure thing @concavelenz will do.

**Comment by Belelros** [comment]
I've just removed the returning from the check on `GECKO` as it doesn't seem to be necessary anymore and they should be all passing AFAIK.

File: closure/goog/style/transform_test.js [hunk 1]
```diff
@@ -100,7 +98,7 @@ function testIsSupported() {
 
 
 function testIs3dSupported() {
-  if (goog.userAgent.GECKO && !goog.userAgent.product.isVersion(10) ||
+  if (!goog.userAgent.product.isVersion(10) ||
```
Comment (shicks) [inline closure/goog/style/transform_test.js]: This doesn't make sense anymore - the right hand side of the || will only run if isVersion(10) but you && it with !isVersion(10), so it's always false.  This is the same thing as simply if (!isVersion(10))

Comment (joeltine) [inline closure/goog/style/transform_test.js]: @Belelros were you going to take a look at this?

Comment (Belelros) [inline closure/goog/style/transform_test.js]: Yep. Will try to get it nailed today.

**Comment by Belelros** [comment]
Any news on this?

File: closure/goog/style/transform_test.js [hunk 2]
```diff
@@ -100,8 +98,7 @@ function testIsSupported() {
 
 
 function testIs3dSupported() {
-  if (goog.userAgent.GECKO && !goog.userAgent.product.isVersion(10) ||
```
Comment (joeltine) [inline closure/goog/style/transform_test.js]: What was this removal of GECKO for?

**Comment by shicks** [comment]
Looks reasonable to merge, except for the testIs3dSupported change.  Could you please revert that last test change and then squash your commits onto our master?  (you may need to `git push -f`).

**Comment by Belelros** [comment]
I may be wrong since this went on a long time ago in a galaxy far far away but I that `GECKO` check is removed so the test can be run against firefox (which was being excluded since it wouldn't work) since the fix is to make it work on Firefox, the test should be run against firefox too. Those are my two cents.

**Comment by joeltine** [comment]
Oh, nah. That test is testing `is3dSupported` which is false for GECKO < 10. The only reason the tests didn't fail is because we don't run them on older GECKO versions.

**Comment by Belelros** [comment]
Fair enough!

**Comment by rzhw** [comment]
Ping, any updates on this?

**Comment by shicks** [comment]
Please squash your commits into a single commit, then rebase it on top of master.  You'll need to `git push -f` to rewrite your fork's history.  Then we can merge.

**Comment by shicks** [comment]
I tried testing this internally and am getting some errors in Firefox:

```
ERROR in testScale
An invalid or illegal string was specified
> goog.style.transform.getScale at http://vwel10.prod.google.com:17273/google3/javascript/closure/style/transform.js:107:18
> setAndAssertScale at http://vwel10.prod.google.com:17273/google3/javascript/closure/style/transform_test.js:79:17
> testScale at http://vwel10.prod.google.com:17273/google3/javascript/closure/style/transform_test.js:176:3

ERROR in testScaleZ
An invalid or illegal string was specified
> goog.style.transform.getScale at http://vwel10.prod.google.com:17273/google3/javascript/closure/style/transform.js:107:18
> setAndAssertScale at http://vwel10.prod.google.com:17273/google3/javascript/closure/style/transform_test.js:79:17
> testScaleZ at http://vwel10.prod.google.com:17273/google3/javascript/closure/style/transform_test.js:172:3
```

**Comment by shicks** [comment]
Closing this as there have been no responses for 2+ years.  Please reopen if you want to continue pursuing it.

**Pull request closed without merging**